plugins/inputs/aerospike/aerospike.go

```diff
@@ -161,6 +161,60 @@ func (a *Aerospike) gatherServer(hostport string, acc telegraf.Accumulator) erro
 			}
 			acc.AddFields("aerospike_namespace", nFields, nTags, time.Now())
 		}
+
+		latency, err := as.RequestNodeInfo(n, "latency:")
+		if err != nil {
+			return err
+		}
+		result, exists := latency["latency:"]
+		if !exists {
+			return errors.New("Aerospike Input Plugin : No key `latency:` in response of as.RequestNodeInfo(n, \"latency:\") from Aerospike Server")
+		}
+		points := strings.Split(result, ";")
+		i := 0
+		for i < len(points) {
+			point := points[i]
+			if strings.Compare(point, "error-no-data-yet-or-back-too-small") == 0 {
+				i++
+			} else {
+				indexOfColon := strings.Index(point, ":")
+				if indexOfColon == -1 {
+					i++
+					continue
+				}
+				histogram := point[0:indexOfColon]
+				operation, namespace := splitNamespaceAndOperation(histogram)
+				latencyKey := point[indexOfColon:]
+				latencyValue := points[i+1]
+				keys := strings.Split(latencyKey, ",")
+				values := strings.Split(latencyValue, ",")
+				ltags := map[string]string{
+					"aerospike_host": hostport,
+					"node_name":      n.GetName(),
+					"namespace":      namespace,
+				}
+				lfields := make(map[string]interface{})
+				var duration, totalOperations float64
+				duration = getDurationOfTransaction(keys[0], values[0])
+				operationsPerSecond, err := parseValue(values[1])
+				if err == nil {
+					totalOperations = operationsPerSecond.(float64) * duration
+					for i := 2; i < len(keys); i++ {
+						metricKey := "latency_" + operation + keys[i]
+						metricValue, err := parseValue(values[i])
+						if err == nil {
+							lfields[metricKey] = metricValue.(float64) * totalOperations
+						} else {
+							log.Printf("I! skipping aerospike field %v with float64 overflow: %q", keys[i], values[i])
+						}
+					}
+				} else {
+					log.Printf("I! skipping aerospike field %v with float64 overflow: %q", keys[1], values[1])
+				}
+				acc.AddFields("aerospike_latency", lfields, ltags, time.Now())
+				i = i + 2
```

Comment:
Interesting API response, do you mind pasting sample output? I'm extremely curious now, thanks.

Reply:
You can check in Readme.md file itself.
eg., aerospike_latency,aerospike_host=localhost:3000,cluster=cache,host=Aerospike-Cache,namespace=test,node_name=BB93B8365DD8E06,service=telegraf latency_read-ops=24358,latency_read>1ms=2.4358,latency_read>8ms=0,latency_read>64ms=0 1538725061000000000